### PR TITLE
port the subchandra problem to C++

### PR DESCRIPTION
**Pull request by zingale**
## PR summary
    this also fixes a few issues:
    the small density cutoff has too high
    the temperature perturbation was not convolving the X(He) correctly
    the runtime parameters were not in the namelist

## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Exec/science/subchandra/problem_setup.H [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+    // initial velocities = 0
+
+    state(i,j,k,UMX) = 0.0_rt;
+    state(i,j,k,UMY) = 0.0_rt;
+    state(i,j,k,UMZ) = 0.0_rt;
+
+
+    int i_he4 = network_spec_index("helium-4");
```
Comment (maxpkatz) [inline Exec/science/subchandra/problem_setup.H:96]: Note my comment from #1331 that the GPU build won't work here.

File: Exec/science/subchandra/problem_setup.H [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    // we don't need to refill xn, since it still holds unchanged from above
+
+    eos(eos_input_rt, eos_state);
+
+    // the internal energy changed
+
+    state(i,j,k,UEINT) = eos_state.e * state(i,j,k,URHO);
+    state(i,j,k,UEDEN) = eos_state.e * state(i,j,k,URHO);
+
+    std::cout << state(i,j,k,URHO) << std::endl;
```
Comment (maxpkatz) [inline Exec/science/subchandra/problem_setup.H:133]: Debug print

File: Exec/science/subchandra/problem_setup.H (hunk 1 above)
Comment (zingale) [reply Exec/science/subchandra/problem_setup.H]: fixed

File: Exec/science/subchandra/problem_setup.H (hunk 2 above)
Comment (zingale) [reply Exec/science/subchandra/problem_setup.H]: fixed

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**